docs/04-reference/wingcli.md

```diff
@@ -184,3 +184,81 @@ pass | test_bucket.w | root/test:put
 ```
 
 We will see that both functions were invoked and that the tests passed.
+
+## CLI Architecture
+
+### CLI - compile explained
+
+The Wing CLI command `compile` is responsible for translating the Wing application from a `*.w` to a target that then can be run and either deploy resources to the cloud or run in the simulator or console.
+
+The `compile` command has three phases.
+
+```mermaid
+flowchart LR
+direction LR
+	Cli --> CompilerGraph --> GeneratePreflightGraph
+subgraph Cli["Prepare"]
+	validate_input["Validate cli arguments"] --> create_work_dir["Create working directories for assets"] --> create_wing_compiler_instance["Create instance of Wing compiler"]
+end
+
+subgraph CompilerGraph["Compile"]
+	direction TB
+    verify_input["Load files and create target folder"] --> wing_file_to_ast["Translate wing file to AST"]--> wing_it["Wing it: Translate Wing-AST to generic AST"] --> type_check_ast["Run type checking"] --> transpile_ast_to_preflight["Transpile AST to preflight program"] --> generate_preflight_program["Write preflight program to disk"]
+end
+
+subgraph GeneratePreflightGraph["Generate target"]
+	direction TB
+    read_preflight_program["Read the preflight program"] --> execute_preflight_program["Execute the preflight program"]
+end
+
+```
+
+**Prepare**
+
+The first one is the `prepare` phase. Here the input arguments of the user are validated and folders for the generated target assets are created. Once this is done an instance of the Wing compiler ist created.
+The Wing compiler is a Rust application that is compiled to a WebAssembly file (`wingc.wasm`). The WebAssembly expose methods that can be invoked from the Wing Cli nodeJS program.
+After the Compiler instance is initialised the Wing application file is parsed to the compiler `compile` method.
+
+**Compile**
```

Comment:
@RaphaelManke I thought perhaps skimming over this https://docs.google.com/presentation/d/1pBen6kYdicl9UiUzzN3ipjfvrrfBbW-WKM2jQmWRjFE/edit#slide=id.g16ba1058826_0_0 can help. It seems you got this pretty much right. But maybe some tweaks in terminology appearing in those slides can help.